kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ZooKeeperACLProvider.scala

```diff
@@ -33,11 +33,20 @@ class ZooKeeperACLProvider(conf: KyuubiConf) extends ACLProvider {
    */
   override lazy val getDefaultAcl: java.util.List[ACL] = {
     val nodeAcls = new java.util.ArrayList[ACL]
-    if (conf.get(HighAvailabilityConf.HA_ZK_ACL_ENABLED)) {
+
+    def addACL(): Unit = {
       // Read all to the world
       nodeAcls.addAll(ZooDefs.Ids.READ_ACL_UNSAFE)
       // Create/Delete/Write/Admin to the authenticated user
       nodeAcls.addAll(ZooDefs.Ids.CREATOR_ALL_ACL)
+    }
+
+    if (conf.get(HighAvailabilityConf.HA_ZK_ACL_ENABLED) &&
+      conf.get(HighAvailabilityConf.HA_ZK_ENGINE_REF_ID).isEmpty) {
```

Comment:
@ulysses-you does this sufficient to distinguish between the server side and the engine side?

Reply:
I don’t seem to find other ways to distinguish server and engine, or use main class to distinguish.